### PR TITLE
Fix MD syntax, spelling, grammar, links, formatting in README.md

### PR DESCRIPTION
**Pull request by doctorlard**
- General copy edit for readability, remove repetition

### CODE REVIEW

**Comment by danmarsden** [comment]
Thanks @doctorlard - I can see some nice improvements there. - few things..

Can you please add back the "branches" header above the version table? - we're trying to standardise that across our plugins eg:
https://github.com/catalyst/moodle-tool_objectfs#branches

Also - it appears a few links to auth_saml (not auth_saml2) have been removed accidentally in the plugins-that-require-simplesamlphp section.

I'm also unsure about the conversion of some of the links / hiding the url and only showing the label - I think it's useful for people to see where they are being directed (especially for links that go to the moodle.org site/plugins db.)

File: README.md [hunk 1]
```diff
@@ -39,95 +37,84 @@ Why is it better?
...
+* IdP initiated flow / IdP first flow / IdP unsolicited logins, eg:
+  http://idp.local/simplesaml/saml2/idp/SSOService.php?spentityid=http://moodle.local/auth/saml2/sp/metadata.php&RelayState=http://moodle.local/course/view.php?id=2
 
 
 Features not yet implemented:
 
 * Enrolment - this should be an enrol plugin and not in an auth plugin
 * Role mapping - not yet implemented
 
-Branches
```
Comment (doctorlard) [inline README.md:71]: Hah, weird... missing heading was probably a missed rebase resolve

File: README.md [hunk 2]
```diff
@@ -389,94 +383,82 @@ function update_attributes($attributes) {
...
 
-* /auth/oauth2
+* `auth/oauth2`
 
-OAuth2 has direct support in Moodle
+    OAuth2 has direct support in Moodle.
 
 
-### Plugins that require SimpleSamlPHP ###
+### Plugins that require SimpleSamlPHP
```
Comment (doctorlard) [inline README.md:415]: Hi @danmarsden - what exactly do you mean by "links to auth_saml (not auth_saml2) have been removed accidentally"

**Comment by doctorlard** [comment]
> I'm also unsure about the conversion of some of the links / hiding the url and only showing the label - I think it's useful for people to see where they are being directed (especially for links that go to the moodle.org site/plugins db.)

This is after all the whole point of Markdown though - the link is visible in the plain text, and presented like normal links when rendered in HTML.

File: README.md (hunk 2 above)
Comment (danmarsden) [reply README.md:415]: if you view the page in github:
https://github.com/catalyst/moodle-auth_saml2/blob/aa47f5ed3c1e9f474b155daad51428859f84395b/README.md

the first item under "Plugins that require SimpleSamlPHP" "auth_saml" is no longer clickable.

**Comment by danmarsden** [comment]
> > I'm also unsure about the conversion of some of the links / hiding the url and only showing the label - I think it's useful for people to see where they are being directed (especially for links that go to the moodle.org site/plugins db.)
> 
> This is after all the whole point of Markdown though - the link is visible in the plain text, and presented like normal links when rendered in HTML.

ignore that it's markdown - 95% of people will be previewing the readme directly within Github. - the context of where the link is taking you to, is just as important as the title IMO - I shouldn't have to click on the link to realise that it's taken me back to moodle.org.

File: README.md (hunk 2 above)
Comment (doctorlard) [reply README.md:415]: Ah! Doh...

**Comment by danmarsden** [comment]
thanks - looks a lot tidier now! :-)

**Comment by brendanheywood** [comment]
+10 @doctorlard thanks for fixing all my garbage typos from forever ago :)

**Comment by doctorlard** [comment]
> +10 @doctorlard thanks for fixing all my garbage typos from forever ago :)
Heh - no worries! :-)

**Pull request merged**